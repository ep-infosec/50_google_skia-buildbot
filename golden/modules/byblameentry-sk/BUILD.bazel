load("//infra-sk:index.bzl", "karma_test", "sk_demo_page_server", "sk_element", "sk_element_puppeteer_test", "sk_page", "ts_library")

sk_element(
    name = "byblameentry-sk",
    sass_deps = [
        "@npm//:node_modules/elements-sk/colors.scss",
    ],
    sass_srcs = ["byblameentry-sk.scss"],
    ts_deps = [
        "//golden/modules:rpc_types_ts_lib",
        "//golden/modules:settings_ts_lib",
        "//infra-sk/modules/ElementSk:index_ts_lib",
        "@npm//common-sk",
        "@npm//elements-sk",
        "@npm//lit-html",
        "//golden/modules:common_ts_lib",
    ],
    ts_srcs = [
        "byblameentry-sk.ts",
        "index.ts",
    ],
    visibility = ["//visibility:public"],
)

sk_page(
    name = "byblameentry-sk-demo",
    html_file = "byblameentry-sk-demo.html",
    sk_element_deps = [":byblameentry-sk"],
    ts_deps = [
        "//golden/modules:settings_ts_lib",
        ":test_data_ts_lib",
    ],
    ts_entry_point = "byblameentry-sk-demo.ts",
)

sk_element_puppeteer_test(
    name = "byblameentry-sk_puppeteer_test",
    src = "byblameentry-sk_puppeteer_test.ts",
    sk_demo_page_server = ":demo_page_server",
    deps = [
        "//puppeteer-tests:util_ts_lib",
        "@npm//@types/chai",
        "@npm//chai",
    ],
)

karma_test(
    name = "byblameentry-sk_test",
    src = "byblameentry-sk_test.ts",
    deps = [
        ":byblameentry-sk",
        ":test_data_ts_lib",
        "//golden/modules:rpc_types_ts_lib",
        "//golden/modules:settings_ts_lib",
        "//infra-sk/modules:test_util_ts_lib",
        "@npm//@types/chai",
        "@npm//@types/sinon",
        "@npm//chai",
        "@npm//common-sk",
        "@npm//sinon",
    ],
)

sk_demo_page_server(
    name = "demo_page_server",
    sk_page = ":byblameentry-sk-demo",
)

ts_library(
    name = "test_data_ts_lib",
    srcs = ["test_data.ts"],
    visibility = ["//visibility:public"],
    deps = ["//golden/modules:rpc_types_ts_lib"],
)
