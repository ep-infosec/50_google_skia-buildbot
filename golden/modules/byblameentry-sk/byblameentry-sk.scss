@use 'node_modules/elements-sk/colors.scss';

byblameentry-sk {
  display: block;

  .blame {
    border: 1px solid var(--light-gray);
    margin: 10px;
    padding: 10px;
  }

  table.affected-tests {
    th {
      white-space: nowrap;
      padding: 5px;
    }
    td {
      padding: 5px;
    }
  }

  ul.blames li {
    margin-bottom: 1em;
  }
}