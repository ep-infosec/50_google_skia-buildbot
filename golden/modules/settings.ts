/**
 * This file lists helpers for returning global constants in Gold. These should not change once
 * an instance has booted up, so they need not be returned via JSON requests.
 *
 * Settings are expected to be in window.GoldSettings and the functions here are nice helpers
 * for that, so as to demystify "where do these values come from?"
 */

export interface GoldSettings {
  title?: string;
  defaultCorpus?: string;
  baseRepoURL?: string;
}

function getSettings(): GoldSettings | undefined {
  return (window as any).GoldSettings as GoldSettings | undefined;
}

export function title(): string {
  return getSettings()?.title || '';
}

export function defaultCorpus(): string {
  return getSettings()?.defaultCorpus || '';
}

export function baseRepoURL(): string {
  return getSettings()?.baseRepoURL || '';
}

export function testOnlySetSettings(newSettings: GoldSettings) {
  (window as any).GoldSettings = newSettings;
}
