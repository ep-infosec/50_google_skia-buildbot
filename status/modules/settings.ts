/**
 * This file lists helpers for returning global constants in Status. These should not change once
 * an instance has booted up, so they need not be returned via JSON requests.
 *
 * Settings are set via <script> on pages in status/pages, filled in by serverside template.
 */

class StatusSettings {
  public swarmingUrl: string = '';

  public treeStatusBaseUrl: string = '';

  // Url with '{{TaskID}}' as a placeholder.
  public logsUrlTemplate: string = '';

  public taskSchedulerUrl: string = '';

  public defaultRepo: string = '';

  public repos: Map<string, string> = new Map();
}

function settings(): StatusSettings | undefined {
  return (<any>window).StatusSettings;
}

// swarmingUrl: Base URL for linking to swarming task data.
export function swarmingUrl() {
  return settings()?.swarmingUrl;
}

// treeStatusBaseUrl: Base URL for getting tree status of specific repos.
export function treeStatusBaseUrl() {
  return settings()?.treeStatusBaseUrl;
}

// taskSchedulerUrl: Base URL for linking to Task Scheduler data.
export function taskSchedulerUrl() {
  return settings()?.taskSchedulerUrl;
}

// logsUrl: Returns a logsUrl for the given taskId.
export function logsUrl(taskId: string): string {
  const temp = settings()?.logsUrlTemplate;
  if (!temp) {
    return '#';
  }
  if (temp.includes('annotations')) {
    if (!taskId.endsWith('0')) {
      return '#';
    }
    // Hack because chromium logs replaces a persistent trailing '0' with a '1' for log urls.
    taskId = `${taskId.slice(0, -1)}1`;
  }
  return temp.replace('{{TaskID}}', taskId);
}

// defaultRepo: Repo to use on initial load.
export function defaultRepo() {
  return settings()?.defaultRepo || '';
}

// repos: List of available repos.
export function repos() {
  const r = settings()?.repos;
  return r ? [...r.keys()] : [];
}

export function repoUrl(r: string) {
  return settings()?.repos.get(r) || '';
}

// revisionUrlTemplate: Returns the base url for a repo's revisions. Can be
// concatenated with a hash to form a valid url.
export function revisionUrlTemplate(repo: string) {
  return settings()?.repos.get(repo);
}

// SetTestSettings: Inject setting values for tests.
export function SetTestSettings(s: StatusSettings) {
  (<any>window).StatusSettings = s;
}
