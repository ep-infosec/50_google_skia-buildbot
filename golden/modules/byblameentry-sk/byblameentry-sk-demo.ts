import './index';
import { entry, fakeNow } from './test_data';
import { testOnlySetSettings } from '../settings';
import { ByBlameEntrySk } from './byblameentry-sk';

Date.now = () => fakeNow;
testOnlySetSettings({
  baseRepoURL: 'https://skia.googlesource.com/skia.git',
});

const byBlameEntrySk = new ByBlameEntrySk();
byBlameEntrySk.byBlameEntry = entry;
byBlameEntrySk.corpus = 'gm';
document.body.appendChild(byBlameEntrySk);
