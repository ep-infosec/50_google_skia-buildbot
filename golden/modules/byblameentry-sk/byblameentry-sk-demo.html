<!DOCTYPE html>
<html>
  <head>
    <title>byblameentry-sk</title>
    <meta charset="utf-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  </head>
  <body class="body-sk">
    <!-- A byblameentry-sk element will be inserted here dynamically. -->
  </body>
</html>
